File: designpattern/src/main/java/com/seniorlibs/designpattern/ch25v2/viewer/ConsoleViewer.java
package com.seniorlibs.designpattern.ch25v2.viewer;

import com.google.gson.Gson;
import com.seniorlibs.designpattern.ch25v2.model.RequestStat;

import java.util.Map;

/**
 * 负责将统计结果显示到命令行中
 */
public class ConsoleViewer implements StatViewer {
    @Override
    public void output(Map<String, RequestStat> requestStats, long startTimeInMillis, long endTimeInMills) {
        System.out.println("Time Span: [" + startTimeInMillis + ", " + endTimeInMills + "]");
        Gson gson = new Gson();
        System.out.println(gson.toJson(requestStats));
    }
}

File: designpattern/src/main/java/com/seniorlibs/designpattern/principle/frame/ch25v2/viewer/ConsoleViewer.kt
package com.seniorlibs.designpattern.principle.frame.ch25v2.viewer

import com.google.gson.Gson
import com.seniorlibs.designpattern.principle.frame.ch25v2.model.RequestStat

/**
 * 负责将统计结果显示到命令行中
 */
class ConsoleViewer : StatViewer {
    override fun output(
        requestStats: Map<String?, RequestStat?>?,
        startTimeInMillis: Long,
        endTimeInMills: Long
    ) {
        println("Time Span: [$startTimeInMillis, $endTimeInMills]")
        val gson = Gson()
        println(gson.toJson(requestStats))
    }
}